File: api/check_order.js
const fetch = require('node-fetch');

module.exports = async (req, res) => {
  const { email, order_id, limit = 1 } = req.query;

  const SHOPIFY_STORE = process.env.SHOPIFY_STORE;
  const SHOPIFY_ACCESS_TOKEN = process.env.SHOPIFY_ACCESS_TOKEN;

  if (!SHOPIFY_STORE || !SHOPIFY_ACCESS_TOKEN) {
    return res.status(500).json({ error: "Missing Shopify config." });
  }

  if (!email && !order_id) {
    return res.status(400).json({ error: "Missing email or order ID" });
  }

  const headers = {
    'X-Shopify-Access-Token': SHOPIFY_ACCESS_TOKEN,
    'Content-Type': 'application/json'
  };

  let url = '';
  if (email) {
    url = `https://${SHOPIFY_STORE}/admin/api/2023-10/orders.json?email=${encodeURIComponent(email)}&status=any&limit=${limit}&order=created_at desc`;
  } else {
    url = `https://${SHOPIFY_STORE}/admin/api/2023-10/orders.json?name=${encodeURIComponent(order_id)}&status=any`;
  }

  try {
    const response = await fetch(url, { headers });
    const data = await response.json();

    if (!data.orders || data.orders.length === 0) {
      return res.status(404).json({ status: "No order found" });
    }

    // If multiple orders returned (via email)
    if (email) {
      const summaries = data.orders.map(order => ({
        order_id: order.name,
        created_at: order.created_at,
        status: order.fulfillment_status || "Unfulfilled",
        tracking: order.fulfillments?.[0]?.tracking_number || "Not yet shipped"
      }));

      return res.status(200).json({
        count: summaries.length,
        orders: summaries
      });
    }

    // If order_id was used, return single order
    const order = data.orders[0];
    const status = order.fulfillment_status || "Unfulfilled";
    const tracking = order.fulfillments?.[0]?.tracking_number || "Not yet shipped";

    return res.status(200).json({
      order_id: order.name,
      status,
      tracking
    });

  } catch (err) {
    return res.status(500).json({ error: "Shopify API error", details: err.message });
  }
};

File: api/check_order.py
import os
import requests

def handler(request):
    email = request.args.get("email")
    order_id = request.args.get("order_id")

    SHOPIFY_STORE = os.getenv("SHOPIFY_STORE")
    SHOPIFY_ACCESS_TOKEN = os.getenv("SHOPIFY_ACCESS_TOKEN")

    if not email and not order_id:
        return {"statusCode": 400, "body": "Missing email or order ID"}

    headers = {
        "X-Shopify-Access-Token": SHOPIFY_ACCESS_TOKEN,
        "Content-Type": "application/json"
    }

    if email:
        url = f"https://{SHOPIFY_STORE}/admin/api/2023-10/orders.json?email={email}"
    else:
        url = f"https://{SHOPIFY_STORE}/admin/api/2023-10/orders.json?name={order_id}"

    response = requests.get(url, headers=headers)

    if response.status_code != 200:
        return {"statusCode": 500, "body": "Shopify API error"}

    orders = response.json().get("orders", [])
    if not orders:
        return {"statusCode": 404, "body": "No order found"}

    order = orders[0]
    status = order["fulfillment_status"] or "Unfulfilled"
    tracking = (
        order.get("fulfillments", [{}])[0].get("tracking_number", "Not yet shipped")
    )

    return {
        "statusCode": 200,
        "body": {
            "order_id": order["name"],
            "status": status,
            "tracking": tracking
        }
    }
